### ISSUE TITLE
`vue/no-useless-mustaches` auto-fix produces invalid HTML

### ISSUE THREAD

**Issue by claneo** [vuejs/eslint-plugin-vue#2384]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.56.0
- **eslint-plugin-vue version:** 9.20.1
- **Vue version:** 
- **Node version:** 
- **Operating System:** 

**Please show your full configuration:**

```json5
module.exports = {
    extends: ['plugin:vue/recommended'],
    rules: {
        'vue/no-useless-mustaches': 'error',
    },
};
```

**What did you do?**	

```vue
<template>
    <span>{{ '<' }}</span>
</template>
```

**What did you expect to happen?**

Don't report error

Or auto fix to html entities

```vue
<template>
    <span>&lt;</span>
</template>
```

**What actually happened?**

```vue
<template>
  <span><</span>
  <!--  ^ Invalid -->
</template>
```

**Repository to reproduce this issue**

[Playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJyNV8uSnDoM/ZVUb7KJa6ruMjWVL8nGGAHOGJtrm35kav79HpnGYB5zZ9XNkZAt6ejB+0W5mi4/L6+R+sHISL9+22/fXsMg7a/39++v3z8+Xl/S02/7+pKVLj8ufjQULj/fL9eRXpTre7JR1NqTivoKk//8SJI/4S7GQEFcpQ8Z7UcTtbg5Xwu8OjjLL1vZ06JinZDeu5toRguTzgahrbjJqLpCJzysYgnbGSPVYvBuIB91aUt12vBhNuKoQrB/j83VMsq1Wk0DXJOsyCLhqj/wFLAy0ku+35k2hejdA/+Mbkg9lKEzTWdgC2oBdyQfhBz056pB4cp1MC5+pkxXuPyZQqNNJH8mfIZfGjGn/0Szi70RZCgxQYcTLW2NtvR/tjgZAUAjwRQRu1Nz3iF5HDT7JqJs04snqilaQsbodTWenszR/JrS1+xdRaUt3gBNTzVOvbsKZ1EXXFGid7Vu9GmeJtWxrxCNWfXU7EhcCo1uxRs9uAWUmiP8gmAPXpHfQEI3G4nRbHiJRfEm3QfnIxdVUF4PCBzFcdhqwKxsOJXyJnVRobluzhT6MSJEdkp+cfSApsMCQicpAuepQesQXPHS1qUkkL/Sp40p62yDlAW7m4RO+sLotsEEXZOgBq7G8IV+NtcPX6HE7xEnSQQlRKm6gpajRSteXSIcCIs+zXCg3V0eyMkbGEmxc/VWGXRuxE3HDv+aMuoQY2iE5fLHhEnvFR4yt1MHL9XYWfykxuPs4tdaK02MQ+agHBqAuk/0bLzruTCy2NO/I8bZKmWrMp2FHBARH0OqJzT6UcWVy7OWJ1vjJA/ee7uTPvt3WCbdTiW6tgX/tU0sieBs0MXMmRWn0K05kSXS6Dp31KJNTvc6It3jQId6HcNkTq6dnXKPNNBdqiXK07k50amYNkJcikdCMnwsKstpEllOftTqbYNn3njntteYmvEOVMbJrZl9q1vjO3DN9BljXu7AA4N669oVpdW7PYhGvTe4YsECqf0NQfcdFjp324PmIG5caglc8VwEzITJbWbdc/l73qelGEvO520N2ew1Ykt3UuNMIV7FJMDVRhWywMjAFRJ0m00xVdHoQINW/CXvFhzTdquKKx9hEkXN+kstTRIsbAY3b+k+gzVVY9uSX54NRS6qBeH5KH0bCmC6ObJZzdN4lqzZtZu5xVTlsMwouP0syvkBfRHhUhzrdFgpHCRnIftG900YkFUvReWcIWn5pDhLGmlM7LBetV2G0J8277fGVdgMS/DZUTegxU57mF1tJ5KlgA8Z9Xgeecu9dRoDYpAqR8E4BNU1TGmlwyp3oBUcqXnkn4TF0g27WF85kxFnnYpwAnfTPX4pKDnks7DoIy7GZAtJeX4AlaNL/b8asUXrxStM++QrLQDcE8mPrIR52GziFI7qBttDjU8hz98SWnGDXnYSFvOew+zzciEQL82iIlQmrQo1DXp01eUBMadkMn2U8XK+yLBIII6VWUFBNoQvBnwPmMcGRSnKdSU9YTc8vx9gSlvkZpGnjcPIipbwbteQ1dpQSfWWZjbZhQuzUOUvwxVYppKXkvR/nokPTc+VgoeXDlY+e+k0SJBWV2y6s9l5t+Ibfnz8B0t8UnM=)
